microcosm_logging/decorators.py

```diff
@@ -15,3 +15,54 @@ class MyClass(object):
     """
     obj.logger = getLogger(obj.__name__)
     return obj
+
+
+class ContextLogger(LoggerAdapter):
+    """
+    Allows for inserting additional context into log records based on the current context.
+    To play nicely with the pythonjsonlogger used for json logs, a ContextLogger will
+    update keys on the existing message object if the message is a dictionary otherwise,
+    it will prepend context information before the log message in brackets.
+
+    """
+    def process(self, msg, kwargs):
+        if isinstance(msg, dict):
+            msg.update(self.extra)
+        else:
+            headers_string = [
```

Comment:
Sure; do we have an easy way to access the current context? The fallback for string formatting is something like:

```
logger.info("foo: {}".format(extras.bar))
```